Sync the OWNERS file with the other repositories

gibizer opened the pull request:
All the other repositories has the structure that allows approving patches for CI folks and the podified core team alongside with the repo specific approver group.

For some reason placement-operator repo was not synced to this structure when the rest of the repo was synced. But there was a discussion in slack[1] over it. There was some objection over this in slack. Lets try to conclude this now.

[1] https://redhat-internal.slack.com/archives/CQXJFGMK6/p1680527086750569

gibizer commented on controllers/suite_test.go:
```diff
@@ -72,11 +78,29 @@ var _ = BeforeSuite(func() {
+	err = (&PlacementAPIReconciler{
+		Client: k8sManager.GetClient(),
+		Scheme: k8sManager.GetScheme(),
+	}).SetupWithManager(k8sManager)
```
our controller's SetupWithManager declare ownership of MariaDBDatabase and KeystoneService objects. It seems that in the test env this declaration prevents our controller to start the reconcile loop (but no error reported). If I comment out all of them then the reconcile loop is started.

gibizer replied:
I see it now. I'n the suit_test.go we need to add the schema of all APIs our controller depends on / can own. That solves the problem for the standard resources like Job, but does fails on CRDs like keystone, mariadb and routes. I have to figure out how to provide those in the test

olliewalsh commented on README.md:
```diff
@@ -0,0 +1,43 @@
+# API Example
+
+The Operator creates a custom PlacementAPI resource that can be used to create Placement API
+instances within the cluster. Example CR to create an Keystone API in your cluster:
```
nit: s/Keystone/Placement/

gibizer replied:
Done

gibizer commented on Dockerfile:
```diff
@@ -24,7 +24,7 @@ RUN mkdir -p ${DEST_ROOT}/usr/local/bin/
 
 # Build manager
-RUN if [ -f $CACHITO_ENV_FILE ] ; then source $CACHITO_ENV_FILE ; fi ; CGO_ENABLED=0  GO111MODULE=on go build ${GO_BUILD_EXTRA_ARGS} -a -o ${DEST_ROOT}/manager main.go
+RUN if [ -f $CACHITO_ENV_FILE ] ; then source $CACHITO_ENV_FILE ; fi ; export GIT_COMMIT=$(git rev-list -1 HEAD) && echo $GIT_COMMIT && CGO_ENABLED=0  GO111MODULE=on go build -ldflags "-X main.gitBuildHash=$GIT_COMMIT" ${GO_BUILD_EXTRA_ARGS} -a -o ${DEST_ROOT}/manager main.go
```
Sure, I moved this to https://github.com/openstack-k8s-operators/placement-operator/pull/16

gibizer commented on controllers/placementapi_controller_test.go:
```diff
@@ -0,0 +1,143 @@
+)
+
+const (
+	// TODO(gibi): Do we want to test in a realistic namespace like "openstack"?
```
ack

gibizer commented:
Now we have 5 test specs run by the github action automatically. You can see the results for the current PR here https://github.com/openstack-k8s-operators/placement-operator/pull/9/checks#step:8:22

stuggi reviewed (commented):
/lgtm and thanks for doing the initial envtest work!

gibizer commented on controllers/placementapi_controller_test.go:
```diff
@@ -0,0 +1,143 @@
+		It("should have the Spec and Status fields initialized", func() {
+			placementAPIInstance := GetPlacementAPIInstance(placementAPILookupKey)
+			Expect(placementAPIInstance.Spec.DatabaseInstance).Should(Equal("test-db-instance"))
+			// TODO(gibi): Why defaulting does not work?
```
I'm wondering where are those webhooks. In a crc env defaulting works for placement. i.e If I don't provide the DatabaseUser in the CR def, when that CR is created, the DatabaseUser is filled automatically. But I don't see any MutatingWebhookConfiguration or ValidatingWebhookConfiguration Kinds in the placement repository. Also asking the crc env for these Kinds returns nothing relevant:

```
[fedora@gibi-f35 ~]$ oc get MutatingWebhookConfiguration
NAME          WEBHOOKS   AGE
machine-api   2          66d
[fedora@gibi-f35 ~]$ oc get ValidatingWebhookConfiguration
NAME                  WEBHOOKS   AGE
machine-api           2          66d
multus.openshift.io   1          66d

```

abays replied:
We haven't actually added any webhooks yet.  If we have a need for them, we can add them via:

```
operator-sdk create webhook --group placement --version v1beta1 --kind PlacementAPI --programmatic-validation --defaulting
```

...and a then a few tweaks to some of the webhook YAMLS in the `config` dir.

gibizer replied:
OK, so the defaulting is not done by a webhook. So back to square one, why I don't see the default values applied in the envtest environment, while they are applied in a real (crc) env.

gibizer commented on controllers/suite_test.go:
```diff
@@ -50,33 +64,91 @@ func TestAPIs(t *testing.T) {
+			GetCRDDirFromPackage("github.com/openstack-k8s-operators/mariadb-operator@*"),
+			// NOTE(gibi): OpenShift CRDs are even trickier as they are not directly published.
+			// For now we store a copy of the needed ones locally
+			filepath.Join("..", "openshift_crds", "route", "v1"),
```
I proposed to move this to lib-common via https://github.com/openstack-k8s-operators/lib-common/pull/26

gibizer committed:
Sync the OWNERS file with the other repositories

All the other repositories has the structure that allows approving
patches for CI folks and the podified core team alongside with the repo
specific approver group.

For some reason placement-operator repo was not synced to this structure when
the rest of the repo was synced. But there was a discussion in slack[1]
over it. There was some objection over this in slack. Lets try to
conclude this now.

[1] https://redhat-internal.slack.com/archives/CQXJFGMK6/p1680527086750569

gibizer commented:
/hold wait until every member of the former approver group votes

bogdando commented:
lgtm

stuggi commented:
lgtm

mrkisaolamb reviewed (commented):
lgtm

gibizer commented:
/unhold
I noted @SeanMooney intentional abstention so we have a quorum here.

gibizer commented:
/lgtm
